### ISSUE TITLE
I’m a newbie to HA trying to load HACS EvNex integration 

### ISSUE THREAD

**Issue by twp176twp** [hardbyte/ha-evnex#42]
Bear with me as I am relatively new to HA. I have loaded the Evnex integration through HACS on a HA Green box. It seems to authenticate with my Evnex login ok but then I get a repeated error. Can anyone help me?

Logs
This error originated from a custom integration.

Logger: custom_components.evnex
Source: custom_components/evnex/__init__.py:185
integration: Evnex EV Charger (documentation, issues)
First occurred: 7:29:24 PM (12 occurrences)
Last logged: 7:40:35 PM

Unhandled exception while updating evnex info
Traceback (most recent call last):
  File "/config/custom_components/evnex/__init__.py", line 185, in async_update_data
    for connector_brief in charge_point_detail.connectors:
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'connectors'

**Comment by fzrashid** [comment]
Having the same issue here. Were you able to solve it?

**Comment by twp176twp** [comment]
No unfortunately. It doesn’t seem to be supported by the author anymore and I don’t know how to fix it. A pity EvNex doesn’t write an official integration. On 2 Nov 2024, at 09:46, fzrashid ***@***.***> wrote:﻿
Having the same issue here. Were you able to solve it?

**Comment by hardbyte** [comment]
Is it possible to update your version of home assistant? Alternatively you could try an older version of the integration.

 I'm running ha 2024.5.1 and it's working for me but I may have a weird python set up as a developer.

**Comment by twp176twp** [comment]
> Is it possible to update your version of home assistant? Alternatively you could try an older version of the integration.
> 
> 
> 
>  I'm running ha 2024.5.1 and it's working for me but I may have a weird python set up as a developer.

Hi Hardbyte,

Thanks for the reply 
I'm running ha 2024.10.2 and updating to the latest 2024.10.4 now but I suspect something they have done with HA has broken your integration. I don't think I can downgrade my HA version on the box I have. I can try and load an older version of the integration to see if that helps. Cheers

**Comment by twp176twp** [comment]
As I am a newbie I cant seem to find how to get all versions to show up in HACS. I have downloaded from github as a zip but Google has not been a help yet to find out how to load it into HA/HACS

**Comment by fzrashid** [comment]
I get this error when I run the latest alpha version.
![image](https://github.com/user-attachments/assets/3a291e0e-e4e4-4ed7-ab26-6cb1a0fc5112)

**Comment by fzrashid** [comment]
More detail of the second error message

![image](https://github.com/user-attachments/assets/0a922824-6262-4d70-8f12-4802662b14c7)

**Comment by twp176twp** [comment]
Yep with the latest alpha version, I get the following error when I put in my credentials and try and add the device
"Config flow could not be loaded: 500 Internal Server Error Server got itself in trouble" and then it shows "not loaded"

I think I got further with the normal release - I think it actually got past the credentials stage before the error.

I cant seem to find the details logs now, not sure what I did

**Comment by hardbyte** [comment]
In case it isn't clear I'm not affiliated with evnex. I'm a software developer who owns an evnex charger and decided to write this library for myself and put it on the internet free for anyone else to use.

Many core plugins for homeassistant rely on a very popular library called `pydantic` that I heavily use in the evnex library. 
The core ha team is having to update all the core integrations before they can update to the v2 version of pydantic - after more than a year they appear to be getting close. There is a strong chance you can "just" update to pydantic v2 on your homeassistant using `pip` but that's at your own risk.

**Comment by twp176twp** [comment]
Thanks. I totally understood you were unaffiliated and just generously wrote the EvNex application yourself. I worked out it was related to pydantic from the error messages and sought to upgrade but I don’t think pip works on my system. Not sure what the HA Green box runs on but it doesn’t seem to have a python environment or python prompt. Apparently many developers that use pydantic have had their HA stuff broken each time they upgrade HA core.CheersOn 10 Nov 2024, at 20:59, Brian Thorne ***@***.***> wrote:﻿
In case it isn't clear I'm not affiliated with evnex. I'm a software developer who owns an evnex charger and decided to write this library for myself and put it on the internet free for anyone else to use.
Many core plugins for homeassistant rely on a very popular library called pydantic that I heavily use in the evnex library.
The core ha team is having to update all the core integrations before they can update to the v2 version of pydantic - after more than a year they appear to be getting close. There is a strong chance you can "just" update to pydantic v2 on your homeassistant using pip but that's at your own risk.

**Comment by The-Organist** [comment]
> In case it isn't clear I'm not affiliated with evnex. I'm a software developer who owns an evnex charger and decided to write this library for myself and put it on the internet free for anyone else to use.
> 
> Many core plugins for homeassistant rely on a very popular library called `pydantic` that I heavily use in the evnex library. The core ha team is having to update all the core integrations before they can update to the v2 version of pydantic - after more than a year they appear to be getting close. There is a strong chance you can "just" update to pydantic v2 on your homeassistant using `pip` but that's at your own risk.

I did a full backup of my Home Assistant before having a go at installing Exnex 6.0.  Upon installing 6.0, I found the pydantic was still only level 1 and I have the simplest install of Home Assistant and don't have the ability to issue any commands like "pip" in my CLI.

Thankfully, the full backup was able to be restored and I can return to 5.3 (with conf fixes from another thread) and wait for the Home Assistant team to upgrade to pydantic 2 :)

Forewarned was forearmed - thanks team!

**Comment by zlodag** [comment]
Interim fix: https://github.com/hardbyte/ha-evnex/issues/36#issuecomment-2544656736

**Comment by The-Organist** [comment]
Totally agree. So good to have this on the “reliable” path again.

**Comment by hardbyte** [comment]
Excellent!
